src/components/KeyboardCompatibleView/hooks/useKeyboardCompatibleHeight.js

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Dimensions, Keyboard, Platform, StatusBar } from 'react-native';
+
+import { useAppState } from './useAppState';
+
+export const useKeyboardCompatibleHeight = ({
+  enabled,
+  initialHeight,
+  rootChannelView,
+}) => {
+  const appState = useAppState();
+
+  const hidingKeyboardInProgress = useRef(false);
+  const [keyboardOpen, setKeyboardOpen] = useState(false);
+
+  const [height, setHeight] = useState(initialHeight);
+
+  const keyboardDidShow = useCallback(
+    (e) => {
+      if (!enabled) {
+        return;
+      }
+
+      hidingKeyboardInProgress.current = false;
+
+      const keyboardHeight = e.endCoordinates.height;
+
+      if (rootChannelView) {
+        rootChannelView.current.measureInWindow((x, y) => {
+          const { height: windowHeight } = Dimensions.get('window');
+          let finalHeight;
+
+          if (Platform.OS === 'android') {
+            finalHeight =
+              windowHeight - y - keyboardHeight - StatusBar.currentHeight;
+          } else {
+            finalHeight = windowHeight - y - keyboardHeight;
+          }
+          // In case if keyboard was closed in meanwhile while
+          // this measure function was being executed, then we
+          // should abort further execution and let the event callback
+          // keyboardDidHide proceed.
+          if (hidingKeyboardInProgress.current) {
+            return;
+          }
+          setHeight(finalHeight);
+          setKeyboardOpen(true);
+        });
+      }
+    },
+    [enabled, hidingKeyboardInProgress, rootChannelView, setHeight],
+  );
+
+  const keyboardDidHide = useCallback(() => {
+    hidingKeyboardInProgress.current = true;
```

Comment:
so on Android does this ref happen once the keyboard hiding is finished as the event name is keyboardDidHide instead of WillHide. So might make for an odd ref. And other than when we have the keyboardDidShow event triggered, this ref will be true after the keyboard is hidden for the first time, even if "hiding" is not still animating

Reply:
hmm maybe we need different name for this ref.